packages/parser/parse/index.js

```diff
@@ -473,8 +514,20 @@ function applyAliases(token, { tokens, logger, filename, src, node }) {
             }
             const aliasOfID = resolveAlias(token.$value[i][property], { tokens, logger, filename, node, src });
             const { id: aliasID, mode: aliasMode } = parseAlias(token.$value[i][property]);
+            const aliasToken = tokens[aliasOfID];
+
+            if (expectedAliasTypes?.[property] && !expectedAliasTypes[property].includes(aliasToken.$type)) {
```

Comment:
Note: this is already a pretty large and complex function that’s doing a lot, but it’s hard to break this up because it’s handling tokens that may be invalid. In general I do like to keep logic dealing with possibly-invalid syntax centralized in one place like this, otherwise you wind up with “this may be invalid” all over your codebase.

Also because we’re in the middle of doing alias resolution which requires pretty much the whole world loaded here, passing that off to other systems probably isn’t good modularization.

All that said, this code is messy for sure, but it’s handling “the whole world” before it’s normalized and validated. The messiness, if quarantined here, helps the rest of the code remain clearer in purpose and less-concerned with edge cases.